modules/ml/xgboost-model-parser/src/main/antlr4/XGBoostModel.g4

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+grammar XGBoostModel;
+
+@header {
+package org.apache.ignite.ml.xgboost.parser;
+}
+
+YES : 'yes' ;
+NO : 'no' ;
+MISSING : 'missing' ;
+EQ : '=' ;
+COMMA : ',' ;
+PLUS : '+' ;
+MINUS : '-' ;
+DOT : '.' ;
+EXP : 'E' | 'e' ;
+BOOSTER : 'booster' ;
+LBRACK : '[' ;
+RBRACK : ']' ;
+COLON : ':' ;
+LEAF : 'leaf' ;
+INT : (PLUS | MINUS)? [0-9]+ ;
+DOUBLE : INT DOT [0-9]* (EXP INT)?;
+STRING : [A-Za-z_][0-9A-Za-z_]+ ;
+NEWLINE : '\r' '\n' | '\n' | '\r' ;
+LT : '<' ;
+WS : ( ' ' | '\t' )+ -> skip ;
+
+xgValue : DOUBLE | INT ;
+xgHeader : BOOSTER LBRACK INT RBRACK COLON? ;
+xgNode : INT COLON LBRACK STRING LT xgValue RBRACK YES EQ INT COMMA NO EQ INT COMMA MISSING EQ INT ;
+xgLeaf : INT COLON LEAF EQ xgValue ;
+xgTree : xgHeader NEWLINE (
```

Comment:
It's not a valid file because it should start with `xgHeader` anyway. Except that your example looks valid. Only one thing we have to check additionally is node indexes and links, but doesn't look like a task for ANTLR.

Reply:
OK, I forget about header but you should understand that such parser can be rewrote to simple regular expressions

Reply:
Let's cut out this grammar and use regexps? Such approach simplify project structure

Reply:
Don't think that we need to replace correct, maintainable and extendable solution with grammar to unsupportable regexps (_Do you think regexp's are good from maintenance perspective?_).

If you are worried about code generation, I agree, it might be a problem. But it solves after a first build of the module. We can even avoid this "build module" requirement by generating source once and adding it into source tree under source version control.